### DISCUSSION TITLE
Data and Models in the Vertical Slices

### DISCUSSION THREAD

**Discussion by ArvidNyden** [jbogard/ContosoUniversityDotNetCore-Pages#70] [Q&A]
Hi! In the documentation of Vertical Sliced Architecture, it looks very much as the data layer also should be a part of the "Feature - Slices". 
How come that you have separated the data layer : https://github.com/jbogard/ContosoUniversityDotNetCore-Pages/blob/master/ContosoUniversity/Data/SchoolContext.cs.

* Should you not keep the DB configuration within the feature? 
* The Models are also separate, the "Domain/Heart" of the app, should that not be included with the feature as well? 

I've used the "Clean Architecture, CQRS" approach before with Web API and the real only different is that you have moved the Controllers together with the Handler / Command / Queries / Validators, would be nice to see you move the data and domain layer as well.

**Comment by jbogard** [comment]
> Should you not keep the DB configuration within the feature?

No, there's a single database for all features so it's a cross-cutting concern that does not belong to one single feature

> The Models are also separate, the "Domain/Heart" of the app, should that not be included with the feature as well?

No, there's a single "Domain" for the entire application, it's also a cross-cutting concern that does not belong to one single feature. You might be able to swing something that has individual aggregate methods as partial classes living inside a feature, but to me that's abusing the partial class feature.
